chia/simulator/block_tools.py

```diff
@@ -414,6 +433,72 @@ def setup_new_gen(
                 removals,
                 cost,
             )
+        if include_transactions and block_fillrate is not None:
+            # if the caller passed in a fill rate, we want to fill blocks up to that fill percentage
+            # we also use BlockBuilder to compress these transactions as well.
+            assert wallet is not None
+            assert rng is not None
+            assert curr.height >= self.constants.HARD_FORK_HEIGHT  # we need new compression for BlockBuilder
+            # function constants
+            adjusted_max_cost: uint64 = uint64(self.constants.MAX_BLOCK_COST_CLVM * block_fillrate / 100)
+            static_cost: uint64 = uint64(4839648)  # cond + exec cost per sb
+            static_comp_cost: uint64 = uint64(7684000)  # byte cost per sb after compression
+            max_batch_size: int = int(1400 * 0.10 * block_fillrate / 100)  # 10% of the way done with the block
+
+            # start building the block
+            avail_coins: set[Coin] = available_coins.copy()  # don't modify the original set
+            builder: BlockBuilder = BlockBuilder()
+            block_full = False
+            total_cost: uint64 = uint64(0)
+            additions = []
+            removals = []
+
+            batch_bundles: list[SpendBundle] = []
+            batch_removals: list[Coin] = []
+            batch_additions: list[Coin] = []
+            while len(avail_coins) > 0:
+                val_after_next_sb = (len(batch_bundles) + 1) * static_comp_cost + total_cost
```

Comment:
if `static_comp_cost` is really the true execution + condition cost for one transaction, why do you add 1 to `len(batch_bundles)`?